torch_tvm/fuse_concat.cpp

```diff
@@ -0,0 +1,59 @@
+#include "fusion_pass.h"
+#include "operators.h"
+
+using namespace torch::jit;
+
+bool isFusableCatNode(Node* node) {
+  if (node->kind() != aten::cat)
+    return false;
+  if (!node->is_constant(attr::dim))
+    return false;
+
+  auto tensors_node = node->namedInput(attr::tensors)->node();
+  if ((tensors_node->inputs().size() + node->outputs().size()) >
+      128) {
+    return false;
+  }
+  if (tensors_node->kind() != prim::ListConstruct)
+    return false;
+
+  if (tensors_node->output()->uses().size() > 1)
+    return false;
+  return true;
+}
+
+Node* createFusedConcat(Node* node) {
+  AT_ASSERT(node->kind() == aten::cat);
+  Graph* graph = node->owningGraph();
+  Node* list_construct = node->namedInput(attr::tensors)->node();
+  int64_t dim = node->get<int64_t>(attr::dim).value();
+
+  Node* fused_cat = graph->create(prim::FusedConcat, list_construct->inputs())
+                        ->i_(attr::dim, dim);
+  fused_cat->insertBefore(list_construct);
+  fused_cat->output()->copyMetadata(node->output());
+  node->output()->replaceAllUsesWith(fused_cat->output());
+  if (list_construct->output()->uses().empty()) {
```

Comment:
Don't you have to first check this condition on node, destroy that and check it again on list_construct? Or does pytorch remove all the nodes without uses on its own?

Reply:
I think the replaceAllUsesWith do it internally.